### PR TITLE
Test scenario main

### PR DESCRIPTION
**Pull request by ruowan**

### CODE REVIEW

**Review by PhoenixHe-msft — APPROVED** [review]

File: regression/jest.setup-file.ts [hunk 1]
```diff
@@ -0,0 +1 @@
+import "reflect-metadata";
```
Comment (raych1) [inline regression/jest.setup-file.ts:1]: Remove this file.

File: lib/liveValidation/liveValidatorLoader.ts [hunk 2]
```diff
@@ -88,24 +87,16 @@ export class LiveValidatorLoader implements Loader<SwaggerSpec> {
...
+    @inject(TYPES.opts) private opts: LiveValidatorLoaderOption,
+    private jsonLoader: JsonLoader,
+    private swaggerLoader: SwaggerLoader
+  ) {
     setDefaultOpts(opts, {
       transformToNewSchemaFormat: false,
-      loadSuppression: Object.keys(allErrorConstants),
     });
 
-    this.jsonLoader = JsonLoader.create(opts);
```
Comment (raych1) [inline lib/liveValidation/liveValidatorLoader.ts:100]: This needs to be added.

Comment (PhoenixHe-msft) [reply lib/liveValidation/liveValidatorLoader.ts:100]: This one is converted to reflect injection so could be removed

File: lib/liveValidation/testRun.ts [hunk 3]
```diff
@@ -0,0 +1,667 @@
+import { LiveValidator } from "./liveValidator";
```
Comment (raych1) [inline lib/liveValidation/testRun.ts:1]: Can you remove this file?

Comment (ruowan) [reply lib/liveValidation/testRun.ts:1]: removed

File: lib/liveValidation/liveValidatorLoader.ts (hunk 2 above)
Comment (ruowan) [reply lib/liveValidation/liveValidatorLoader.ts:100]: Just pass `schemaValidationOption` to `AjvSchemaValidator`

File: package.json [hunk 4]
```diff
@@ -17,15 +17,21 @@
     "@azure/ms-rest-js": "^2.1.0",
     "@azure/openapi-markdown": "^0.9.2",
     "@microsoft.azure/autorest-extension-base": "1.0.13",
+    "@types/difflib": "^0.2.0",
```
Comment (raych1) [inline package.json:20]: Can you move this to devdenpendencies?

File: lib/liveValidation/liveValidatorLoader.ts [hunk 5]
```diff
@@ -88,24 +87,16 @@ export class LiveValidatorLoader implements Loader<SwaggerSpec> {
...
-  );
-  private constructor(private opts: LiveValidatorLoaderOptions) {
+  public constructor(
+    @inject(TYPES.opts) private opts: LiveValidatorLoaderOption,
+    private jsonLoader: JsonLoader,
+    private swaggerLoader: SwaggerLoader
+  ) {
     setDefaultOpts(opts, {
       transformToNewSchemaFormat: false,
-      loadSuppression: Object.keys(allErrorConstants),
```
Comment (raych1) [inline lib/liveValidation/liveValidatorLoader.ts:97]: How about this one for suppression settings?

Comment (ruowan) [reply lib/liveValidation/liveValidatorLoader.ts:97]: Good catch. loadSuppression is about what kind of error to be suppressed?

**Review by raych1 — APPROVED** [review]

**Review by raych1 — APPROVED** [review]

**Pull request merged**